service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -52,6 +53,11 @@ public void onMicroserviceInstanceHeartbeatTask(MicroserviceInstanceHeartbeatTas
     }
   }
 
+  @Subscribe
```

Comment:
it's better to change task controller
when failed to register instance, then change status, next time, task will run from register microservice

Reply:
how to change task controller ?

Reply:
Just set 
this.taskStatus = TaskStatus.READY;
this.registered = false;

like onMicroserviceInstanceHeartbeatTask

Reply:
get it